extras/build/material-icons.js

```diff
@@ -1,48 +1,103 @@
-const packageName = 'material-design-icons'
+const packageName = ''
 const iconSetName = 'Google Material Design Icons'
 
 // ------------
 
-const glob = require('glob')
-const { readFileSync, writeFileSync } = require('fs')
 const { copySync } = require('fs-extra')
 const { resolve } = require('path')
+const fetch = require('cross-fetch')
 
-let skipped = []
-const distFolder = resolve(__dirname, `../material-icons`)
-const { extract, writeExports } = require('./utils')
+const skipped = {}
+const distFolder = {}
+const iconNames = {}
+const svgExports = {}
+const typeExports = {}
 
-const svgFolder = resolve(__dirname, `../node_modules/${packageName}/`)
-const svgFiles = glob.sync(svgFolder + '/*/svg/production/ic_*_24px.svg')
-const iconNames = new Set()
+const { extractSvg, writeExports, Queue, sleep, retry } = require('./utils')
 
-const svgExports = []
-const typeExports = []
+const themeMap = {
+  baseline: '', // filled
+  outlined: '_outlined',
+  round: '_round',
+  sharp: '_sharp',
+}
 
-svgFiles.forEach(file => {
-  const name = ('mat_' + file.match(/ic_(.*)_24px\.svg/)[1])
-    .replace(/(_\w)/g, m => m[1].toUpperCase())
+function downloadIcon(icon) {
+  return Promise.all(
+    Object.keys(themeMap).map(async (theme) => {
+      const themeName = themeMap[theme]
+      const formattedTheme = themeName.split('_').join('')
+      const response = await fetch(
```

Comment:
Relatively expensive processes such as fetching and processing logic could be moved under the below if statement:
```js
if (iconNames[theme].has(name)) {
  return
}
```
It would be better for performance.

Reply:
fixed...